Bukkit/src/main/java/com/plotsquared/bukkit/queue/LimitedRegionWrapperQueue.java

```diff
@@ -0,0 +1,114 @@
+ *            PlotSquared plot management system for Minecraft
+ *               Copyright (C) 2014 - 2022 IntellectualSites
+ *
+ *     This program is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     This program is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+package com.plotsquared.bukkit.queue;
+
+import com.plotsquared.bukkit.schematic.StateWrapper;
+import com.plotsquared.core.queue.DelegateQueueCoordinator;
+import com.sk89q.jnbt.CompoundTag;
+import com.sk89q.worldedit.bukkit.BukkitAdapter;
+import com.sk89q.worldedit.entity.Entity;
+import com.sk89q.worldedit.function.pattern.Pattern;
+import com.sk89q.worldedit.math.BlockVector3;
+import com.sk89q.worldedit.world.block.BaseBlock;
+import com.sk89q.worldedit.world.block.BlockState;
+import org.bukkit.Location;
+import org.bukkit.entity.EntityType;
+import org.bukkit.generator.LimitedRegion;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+/**
+ * Wraps a {@link LimitedRegion} inside a P2 {@link com.plotsquared.core.queue.QueueCoordinator} so it can be written to.
+ *
+ * @since TODO
+ */
+public class LimitedRegionWrapperQueue extends DelegateQueueCoordinator {
+
+    private final LimitedRegion limitedRegion;
+
+    /**
+     * @since TODO
+     */
+    public LimitedRegionWrapperQueue(LimitedRegion limitedRegion) {
+        super(null);
+        this.limitedRegion = limitedRegion;
+    }
+
+    @Override
+    public boolean setBlock(final int x, final int y, final int z, @NonNull final Pattern pattern) {
+        return setBlock(x, y, z, pattern.applyBlock(BlockVector3.at(x, y, z)));
+    }
+
+    @Override
+    public boolean setBlock(final int x, final int y, final int z, @NonNull final BaseBlock id) {
+        boolean result = setBlock(x, y, z, id.toImmutableState());
+        if (result && id.hasNbtData()) {
+            CompoundTag tag = id.getNbtData();
+            StateWrapper sw = new StateWrapper(tag);
+            sw.restoreTag(limitedRegion.getBlockState(x, y, z).getBlock());
+        }
+        return result;
+    }
+
+    @Override
+    public boolean setBlock(final int x, final int y, final int z, @NonNull final BlockState id) {
+        try {
+            limitedRegion.setType(x, y, z, BukkitAdapter.adapt(id.getBlockType()));
+            limitedRegion.setBlockData(x, y, z, BukkitAdapter.adapt(id));
+        } catch (IllegalArgumentException e) {
+            e.printStackTrace();
```

Comment:
If we can ignore the exception, this should be removed. If we can't ignore it, it should be logged via a logger instead.